contracts/liquidity_hub/pool-manager/src/router/commands.rs

```diff
@@ -161,3 +166,79 @@ pub fn execute_swap_operations(
         ])
         .add_attributes(swap_attributes))
 }
+
+pub fn add_swap_routes(
+    deps: DepsMut,
+    sender: Addr,
+    swap_routes: Vec<SwapRoute>,
+) -> Result<Response, ContractError> {
+    let mut attributes = vec![];
+
+    for swap_route in swap_routes {
+        simulate_swap_operations(
+            deps.as_ref(),
+            Uint128::one(),
+            swap_route.clone().swap_operations,
+        )
+        .map_err(|_| ContractError::InvalidSwapRoute(swap_route.clone()))?;
+
+        // TODO: do we need to derivate the key from the pair identifier too?
+        let swap_route_key =
+            SWAP_ROUTES.key((&swap_route.offer_asset_denom, &swap_route.ask_asset_denom));
```

Comment:
> // TODO: do we need to derivate the key from the pair identifier too?
> Yes I think we do need this, reason being is we designed pool manager to be able to have any amount of Pools with X and Y same assets 
> So there can be for example 3 uwhale-uluna pools all with different settings or even the same settings for this reason good to use pool_identifier here too 

This is correct

Reply:
I think it's enough to build the key with the assets because otherwise you could have multiple swap routes based on the pool identifier and it makes it overkill imo

Reply:
I'll leave it then as it is, removed TODOs